### PR TITLE
Hometask3 arrays

### PR DESCRIPTION
**Pull request by AndriiBakayev**
Hometask 3 Arrays with PSR12 and Doc

### CODE REVIEW

File: src/arrays/Arrays.php [hunk 1]
```diff
@@ -1,57 +1,81 @@
...
-    public function repeatArrayValues(array $input): array{
-        $output=[];
-        foreach($input as $key=>$val){
-            for ($i=0; $i<$val;$i++)
-                $output[]=$val;
+    public function repeatArrayValues(array $input): array
+    {
+        $output = [];
+        foreach ($input as $key => $val) {
+            for ($i = 0; $i < $val; $i++) {
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php]: please try to use php array function

File: src/arrays/Arrays.php [hunk 2]
```diff
@@ -1,57 +1,81 @@
...
+                $output[] = $val;
+            }
         }
-        return $output;   
+        return $output;
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php]: please add empty line before return

File: src/arrays/Arrays.php [hunk 3]
```diff
@@ -1,57 +1,81 @@
...
+    public function getUniqueValue(array $input): int
+    {
+
+        $ret = 0;
+        foreach (array_count_values($input) as $key => $val) {
+            if ($val == 1 && ($key < $ret || $ret == 0)) {
+                $ret = $key;
+            }
+        }
         return $ret;
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php]: please add empty line before return

File: src/arrays/Arrays.php [hunk 4]
```diff
@@ -1,57 +1,81 @@
...
+     * @return int output the least uniqie value or 0 if it not found
      */
-    public function getUniqueValue(array $input): int{
-        $ret=0;
-        foreach (array_count_values($input) as $key=>$val)
-        if ($val==1 && ($key<$ret || $ret==0))
-            $ret=$key;
+    public function getUniqueValue(array $input): int
+    {
+
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php:76]: please remove empty line

File: src/arrays/Arrays.php [hunk 5]
```diff
@@ -1,57 +1,81 @@
...
+        $ret = 0;
+        foreach (array_count_values($input) as $key => $val) {
+            if ($val == 1 && ($key < $ret || $ret == 0)) {
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php]: pleaese use strict comparison ===

File: src/arrays/Arrays.php [hunk 6]
```diff
@@ -71,17 +95,27 @@ public function getUniqueValue(array $input): int{
...
+        foreach ($input as $val) {
+            foreach ($val['tags'] as $tag) {
+                $tagsNames[$tag][] = $val['name']; 
+                //Make Tag => Name(s) associative array
+            }
+        }
+        foreach ($tagsNames as $key => $val) {
+             sort($tagsNames[$key]);
+        } //Internal sort of each element
+        return $tagsNames;
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php:120]: please add empty line before return

File: src/arrays/Arrays.php [hunk 7]
```diff
@@ -71,17 +95,27 @@ public function getUniqueValue(array $input): int{
...
-            // array_multisort 
-        return $nametags;
-    }
+    public function groupByTag(array $input): array
+    {
 
-}
\ No newline at end of file
+        $tagsNames = [];
+        foreach ($input as $val) {
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php]: please try to find different solution

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

**Review by yderevianko-cartera — APPROVED** [review]

File: src/arrays/Arrays.php (hunk 2 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php]: Done

File: src/arrays/Arrays.php (hunk 3 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php]: Done

File: src/arrays/Arrays.php [hunk 8]
```diff
@@ -1,57 +1,90 @@
...
-    public function repeatArrayValues(array $input): array{
+    public function repeatArrayValues(array $input): array
+    {
         $output=[];
-        foreach($input as $key=>$val){
-            for ($i=0; $i<$val;$i++)
-                $output[]=$val;
+        // Newer version - have no mension how to supply link to $output array 
+        // via pointer into function:
+        // array_walk($input,function(&$output,$value, $key, ){
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php]: please remove comments

File: src/arrays/Arrays.php (hunk 4 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php:76]: Done

File: src/arrays/Arrays.php (hunk 1 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php]: Tried 2 more algorithms and found maybe 
the most array-functioned method

File: src/arrays/Arrays.php [hunk 9]
```diff
@@ -1,57 +1,107 @@
...
-            for ($i=0; $i<$val;$i++)
-                $output[]=$val;
-        }
-        return $output;   
+    public function repeatArrayValues(array $input): array
+    {
+
+        array_walk(
+            $input,
+            function (&$val) {
```
Comment (yderevianko-cartera) [inline src/arrays/Arrays.php:47]: Good job! Now you know about php array function more ))

**Review by yderevianko-cartera — APPROVED** [review]

File: src/arrays/Arrays.php (hunk 7 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php]: Can't find except including sort into internal foreach. IMHO not a good idea becaose sorts always have a slow algorithm.
Have any ideas? 
The problem is I can not get array back from callable function back into main function:
in array_walk, array_walk_recursive, array_map :
 - tried to use static variables
 - tried to use gloval variables
 - tried to use &$pointer notation

File: src/arrays/Arrays.php (hunk 8 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php]: Done

**Review by yderevianko-cartera — APPROVED** [review]

File: src/arrays/Arrays.php (hunk 5 above)
Comment (AndriiBakayev) [reply src/arrays/Arrays.php]: Foreach is gone

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**